crates/ruff/Cargo.toml

```diff
@@ -54,6 +54,8 @@ walkdir = { workspace = true }
 wild = { workspace = true }
 
 [dev-dependencies]
+# Enable test rules during development
+ruff_linter = { path = "../ruff_linter", features = ["clap", "test-rules"] }
```

Comment:
@BurntSushi - Can you double-confirm for me that these won't enable in non-`cargo test` builds?

Reply:
```
❯ ./target/release/ruff check example.py --select RUF
❯ ./target/debug/ruff check example.py --select RUF
```

Both do nothing

Reply:
If I do `cargo build --release --all-features` they are present

```
❯ ./target/release/ruff check example.py --select RUF
example.py:1:1: RUF900 Hey this is a stable test rule.
example.py:1:1: RUF901 [*] Hey this is a stable test rule with a safe fix.
example.py:1:1: RUF902 Hey this is a stable test rule with an unsafe fix.
example.py:1:1: RUF903 Hey this is a stable test rule with a display only fix.
Found 4 errors.
[*] 1 fixable with the `--fix` option (1 hidden fix can be enabled with the `--unsafe-fixes` option).
```

but they are not present without `--all-features` in the `release` or `debug` targets.

Reply:
Idk what features we include in releases, it's buried in maturin?

Reply:
We don't specify anything, so we get the default features.

Reply:
(Which should omit these.)

Reply:
Oh great so I think we're good then

Reply:
Yeah features enabled in `dev-dependencies` should definitely not cross over into standard builds. But yeah as Zanie mentioned, if you have anything that's using `--all-features`, then they will get enabled in that case.